src/pipdeptree/_discovery.py

```diff
@@ -64,17 +76,31 @@ def get_installed_distributions(
             already_seen_dists = first_seen_to_already_seen_dists_dict.setdefault(seen_dists[normalized_name], [])
             already_seen_dists.append(dist)
 
-    should_print_warning = warning_printer.should_warn() and first_seen_to_already_seen_dists_dict
-    if should_print_warning:
-        warning_printer.print_multi_line(
-            "Duplicate package metadata found",
-            lambda: render_duplicated_dist_metadata_text(first_seen_to_already_seen_dists_dict),
-            ignore_fail=True,
-        )
+    if warning_printer.should_warn():
+        if site_dir_with_invalid_metadata:
+            warning_printer.print_multi_line(
+                "Missing or invalid metadata found in the following site dirs",
+                lambda: render_invalid_metadata_text(site_dir_with_invalid_metadata),
+            )
+        if first_seen_to_already_seen_dists_dict:
+            warning_printer.print_multi_line(
+                "Duplicate package metadata found",
+                lambda: render_duplicated_dist_metadata_text(first_seen_to_already_seen_dists_dict),
+                ignore_fail=True,
+            )
 
     return dists
 
 
+def has_valid_metadata(dist: Distribution) -> bool:
+    return dist.metadata["Name"] is not None
+
+
+def render_invalid_metadata_text(site_dirs_with_invalid_metadata: set[str]) -> None:
+    for site_dir in site_dirs_with_invalid_metadata:
+        print(site_dir, file=sys.stderr)  # noqa: T201
```

Comment:
There isn't a way to get the invalid distribution metadata directory directly without having to jump to using the private `_path` variable, but we do have the site directory where the invalid metadata resides in so I decided to use that.

Not sure if there is a better way to approach this at the moment.